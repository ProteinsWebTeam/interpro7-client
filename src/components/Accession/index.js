import React, { PureComponent } from 'react';
import T from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import { UniProtLink, ProteomeLink } from 'components/ExtLink';

const Default = ({ children }) => <span>{children}</span>;
Default.propTypes = {
  children: T.oneOfType([T.string, T.number]).isRequired,
};

/*:: type Props = {
  accession: string | number,
  db: string,
}; */

export class Accession extends PureComponent /*:: <Props> */ {
  static propTypes = {
    accession: T.oneOfType([T.string, T.number]).isRequired,
    title: T.string,
    db: T.string,
  };

  render() {
    const { accession, id, db, title } = this.props;
    return (
      <div>
        {title || 'Accession'}: {accession}
      </div>
    );
  }
}

const mapStateToProps = createSelector(
  state =>
    state.customLocation.description[state.customLocation.description.main.key]
      .db,
  db => ({ db }),
);

export default connect(mapStateToProps)(Accession);
