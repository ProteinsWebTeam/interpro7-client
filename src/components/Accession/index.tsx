import React from 'react';

import cssBinder from 'styles/cssBinder';
import local from './style.css';

const f = cssBinder(local);

type Props = {
  accession: string | number;
  withTitle?: boolean;
  title?: string;
};

const Accession = ({ withTitle, accession, title }: Props) => {
  return (
    <div>
      {title !== '' && withTitle ? <span> {title || 'Accession'}:</span> : null}
      {title === 'Job ID' ? (
        <span className={f('tag-sqc')}> {accession}</span>
      ) : (
        accession
      )}
    </div>
  );
};

export default Accession;
