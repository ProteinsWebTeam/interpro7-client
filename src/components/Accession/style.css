.tag-sqc {
  background-color: var(--colors-other-main);
  padding: 0.3rem 0.5rem;
  color: var(--colors-light-title);
}
