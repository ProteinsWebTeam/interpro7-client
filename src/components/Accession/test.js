import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Accession from '.';

const renderer = new ShallowRenderer();

describe('<Accession />', () => {
  test('should render the accession component for UniProt reviewed', () => {
    renderer.render(
      <Accession db="reviewed" accession="A0A023GPI8" id="LECA_CANBL" />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    renderer.render(<Accession db="reviewed" accession="A0A023GPI8" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('should render the accession component for UniProt unreviewed', () => {
    renderer.render(
      <Accession
        db="unreviewed"
        accession="A0A1S2IFJ5"
        id="A0A1S2IFJ5_9MICO"
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    renderer.render(<Accession db="unreviewed" accession="A0A1S2IFJ5" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('should render the accession component for UniProt proteome', () => {
    renderer.render(<Accession db="proteome" accession="UP000000278" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('should render the accession component for taxon', () => {
    renderer.render(<Accession db="taxonomy" accession={1} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('should render the accession component for other', () => {
    renderer.render(
      <Accession
        db="sequence"
        accession="iprscan5-R20171012-151028-0135-99171282-p1m"
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
