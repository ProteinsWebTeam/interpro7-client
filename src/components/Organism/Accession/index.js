// @flow
import React from 'react';
import T from 'prop-types';

import { TaxLink } from 'components/ExtLink';

const Accession = (
  {
    metadata: { accession, id },
  } /*: {metadata: {accession: string | number, id: ?string}} */,
) => (
  <div>
    Accession: <TaxLink id={accession}>{accession}</TaxLink>
    {id && ` (${id})`}
  </div>
);

Accession.propTypes = {
  metadata: T.shape({
    accession: T.oneOfType([T.string, T.number]).isRequired,
    id: T.string,
  }).isRequired,
};

export default Accession;
