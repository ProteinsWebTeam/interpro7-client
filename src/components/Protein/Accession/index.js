// @flow
import React from 'react';
import T from 'prop-types';

import { UniProtLink } from 'components/ExtLink';

import { foundationPartial } from 'styles/foundation';

import ipro from 'styles/interpro-new.css';
const f = foundationPartial(ipro);

const Accession = (
  {
    metadata: { accession, id },
  } /*: {metadata: {accession: string | number, id?: string}} */,
) => (
  <span>
    Accession: <UniProtLink id={accession} className={f('ext')} />
    {id && ` (${id})`}
  </span>
);

Accession.propTypes = {
  metadata: T.shape({
    accession: T.oneOfType([T.string, T.number]).isRequired,
    id: T.string,
  }).isRequired,
};

export default Accession;
