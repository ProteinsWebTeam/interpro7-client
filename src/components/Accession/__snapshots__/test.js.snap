// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accession /> should render the accession component for UniProt proteome 1`] = `
<div>
  Accession
  :
   
  <ProteomeLink
    id="UP000000278"
  >
    UP000000278
  </ProteomeLink>
</div>
`;

exports[`<Accession /> should render the accession component for UniProt reviewed 1`] = `
<div>
  Accession
  :
   
  <UniProtLink
    id="A0A023GPI8"
  >
    A0A023GPI8
  </UniProtLink>
   (LECA_CANBL)
</div>
`;

exports[`<Accession /> should render the accession component for UniProt reviewed 2`] = `
<div>
  Accession
  :
   
  <UniProtLink
    id="A0A023GPI8"
  >
    A0A023GPI8
  </UniProtLink>
</div>
`;

exports[`<Accession /> should render the accession component for UniProt unreviewed 1`] = `
<div>
  Accession
  :
   
  <UniProtLink
    id="A0A1S2IFJ5"
  >
    A0A1S2IFJ5
  </UniProtLink>
   (A0A1S2IFJ5_9MICO)
</div>
`;

exports[`<Accession /> should render the accession component for UniProt unreviewed 2`] = `
<div>
  Accession
  :
   
  <UniProtLink
    id="A0A1S2IFJ5"
  >
    A0A1S2IFJ5
  </UniProtLink>
</div>
`;

exports[`<Accession /> should render the accession component for other 1`] = `
<div>
  Accession
  :
   
  <Default
    id="iprscan5-R20171012-151028-0135-99171282-p1m"
  >
    iprscan5-R20171012-151028-0135-99171282-p1m
  </Default>
</div>
`;

exports[`<Accession /> should render the accession component for taxon 1`] = `
<div>
  Accession
  :
   
  <Default
    id={1}
  >
    1
  </Default>
</div>
`;
